### ISSUE TITLE
 System.Net.Http.WinHttpException: The operation timed out

### ISSUE THREAD

**Issue by Eilon** [dotnet/corefx#35839]
_From @shakibamor on Wednesday, 06 March 2019 20:36:18_

I am testing my code in my win10 machine and works fine.

//myCode
  var req = new HttpRequestMessage(HttpMethod.Get, url);
            var res = await client.SendAsync(req);

I am using .net core 2.0.0 and sdk 2.1.103. 
When i publish the project in a winServer2012 R2
I get below error for my httprequest: 

 [ERR] System.Net.Http.HttpRequestException: An error occurred while sending the request. ---> System.Net.Http.WinHttpException: The operation timed out
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Threading.Tasks.RendezvousAwaitable`1.GetResult()
   at System.Net.Http.WinHttpHandler.<StartRequest>d__105.MoveNext()
   --- End of inner exception stack trace ---

I am sending request to the secured services. I had this issue just for POST request then I migrate to 2.1 and it got worse because this timeout error happened for both GET and POST requests. I switched it back to 2.0. but still all http requests get timeout issue.

_Copied from original issue: aspnet/AspNetCore#8245_

**Comment by davidsh** [comment]
@shakibamor you mentioned that also tried your application was .NET Core 2.1.  .NET Core 2.1 uses a different HTTP stack implementation so you would not ever get 'WinHttpException' being thrown.  What sort of errors did you get with .NET Core 2.1.

Since you got timeout errors with both .NET Core 2.0 and 2.1 it likely the issue is related to something in your network environment and is not specifically a bug in the .NET Core platform.

You mentioned that everything works fine on Windows 10 but starts to fail with Windows Server 2012 R2.  Is that still true.

Do you have a small repro we can run to reproduce the problem and diagnose it?  Or perhaps you can take Wireshark traces etc. Those things will help narrow down the problem.  Otherwise, it is not possible to determine the cause of the problem with just the information presently listed in this issue.

**Comment by shakibamor** [comment]
My issue is solved. When I ran the application in visual studio in my windows 10 I got no error and everything worked fine although I had http calls to the secured services and proxy was involved. Right now I added HttpClient Handler and WebProxy to this handler and my application is working fine in winServer2012 R2 as well. I am not sure why it does not have issue in my windows and through visual studio without this handler. But anyway my application is working fine now.
